### PR TITLE
Remove all SSR tags on mount

### PR DESCRIPTION
**Pull request by TrySound**
Ref https://github.com/tizmagik/react-head/issues/44

In this diff I remove selector builder in favour of querying and
removing all SSR tags.

This should fix the problem with language characters and unescaped
quotes.

### CODE REVIEW

File: src/HeadProvider.js [hunk 1]
```diff
@@ -66,6 +66,14 @@ export default class HeadProvider extends React.Component {
     },
   };
 
+  componentDidUpdate() {
```
Comment (tizmagik) [inline src/HeadProvider.js]: Curious, why in this lifecycle method versus, say, componentDidMount ?

File: src/HeadProvider.js [hunk 2]
```diff
@@ -66,6 +66,14 @@ export default class HeadProvider extends React.Component {
...
+    const ssrTags = document.head.querySelectorAll(`[data-rh=""]`);
+
+    for (let i = 0; i < ssrTags.length; i += 1) {
```
Comment (tizmagik) [inline src/HeadProvider.js]: Could we just forEach here? `ssrTags.forEach(t => t.remove());`

File: src/HeadProvider.js (hunk 1 above)
Comment (TrySound) [reply src/HeadProvider.js]: Just mixed up

File: src/HeadProvider.js (hunk 2 above)
Comment (TrySound) [reply src/HeadProvider.js]: Sure

Comment (TrySound) [reply src/HeadProvider.js]: However, do you want to support ie11?

Comment (TrySound) [reply src/HeadProvider.js]: Or we can support it with the request.

**Pull request merged**